File: LeetCode/Medium/2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        pos = []
        neg = []
        ret = []
        
        for num in nums:
            if num < 0:
                neg.append(num)
            else:
                pos.append(num)
        
        i = 0
        
        while i < len(pos):
            ret.append(pos[i])
            ret.append(neg[i])
            i += 1
        
        return ret

File: LeetCode/Medium/2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int n = nums.size();
        int i = 0;
        int j = 1;
        vector<int> ret(n);
        
        for (int idx = 0; idx < n; ++idx){
            if (nums[idx] > 0){
                ret[i] = nums[idx];
                i += 2;
            } else {
                ret[j] = nums[idx];
                j += 2;
            }   
        }
        
        return ret;
    }
};